command/token_renew.go

```diff
@@ -140,5 +150,10 @@ func (c *TokenRenewCommand) Run(args []string) int {
 		return 2
 	}
 
+	if c.flagFailIfNotFullfilled && secret.LeaseDuration < int(increment.Seconds()) {
+		c.UI.Info("Token renewal failed: requested increment could not be fully fulfilled")
```

Comment:
The big thing I want opinions on is whether it makes sense to implement this on CLI only. As it is now the token is successfully renewed with a capped duration but the output message seems to imply that it wasn't renewed at all, which might be unsafe as the user could be leaving behind a token with expiration longer than expected. Of course that could be made clearer with a more specific message like `Token renewal completed with capped duration, failing the command because of --fail-if-not-fulfilled`, but I'm wondering if it doesn't make more sense for this parameter to make its away to the API so that Vault can reject the request and not renew the token if it cannot fully fulfill the request. Will get back to this once I've gathered folks' thoughts.

Reply:
I brought up this discussion internally and it we're good with adding these proposed changes to the CLI instead of implementing something at the API level. I think we should still change the wording tho to make it clear that the token was renewed but the command is failing because the request wasn't fully fulfilled

Reply:
I will make changes about wording and other requested changes as soon as possible.